### PR TITLE
Instrumentation: log the total number of db queries per request

### PR DESCRIPTION
**Pull request by bergquist**
Signed-off-by: bergquist <carl.bergquist@gmail.com>

### CODE REVIEW

**Review by sakjur — APPROVED** [review]

File: pkg/infra/log/databaseCounter_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCountngDatabaseCalls(t *testing.T) {
+
```
Comment (sakjur) [inline pkg/infra/log/databaseCounter_test.go:11]:
```suggestion
```

File: pkg/infra/log/databaseCounter_test.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+func TestCountngDatabaseCalls(t *testing.T) {
+	ctx := context.Background()
+
+	ctx = IncDBCallCounter(ctx)
+	ctx = IncDBCallCounter(ctx)
+	ctx = IncDBCallCounter(ctx)
+
+	count := TotalDBCallCount(ctx)
+	assert.Equal(t, int64(3), count, "expect counter to increasement three times")
```
Comment (hairyhenderson) [inline pkg/infra/log/databaseCounter_test.go:18]:
```suggestion
func TestCountingDatabaseCalls(t *testing.T) {
	ctx := context.Background()

	ctx = IncDBCallCounter(ctx)
	ctx = IncDBCallCounter(ctx)
	ctx = IncDBCallCounter(ctx)

	count := TotalDBCallCount(ctx)
	assert.Equal(t, int64(3), count, "expect counter to increase three times")
```

File: pkg/infra/log/databaseCounter.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"context"
+	"sync/atomic"
+)
+
+type contextKey struct{}
+
+var dbCallCounterNameKey = contextKey{}
+
+// InitCounter creates a pointer on the context that can be increamented later
```
Comment (hairyhenderson) [inline pkg/infra/log/databaseCounter.go:12]:
```suggestion
// InitCounter creates a pointer on the context that can be incremented later
```

**Review by hairyhenderson — APPROVED** [review]
some minor typos, otherwise LGTM :shipit:

**Pull request merged**